sbpy/activity/dust.py

```diff
@@ -1084,181 +607,36 @@ def fluxd(self, wave_or_freq, aper, eph, Tscale=1.1, T=None, unit=None,
-            or angular units), or as an sbpy `~sbpy.activity.Aperture`
-            class.
-
-        eph : dictionary-like, `~sbpy.data.Ephem`, optional
-            Ephemerides of the comet, describing heliocentric and
-            geocentric distances.
-
-        bandpass : string, `~synphot.SpectralElement`, optional
-            Compute the 0-mag flux density in this bandpass.  Provide
-            either the name of a :mod:`~synphot` filter, or a
-            transmission spectrum.  Ignored if ``fluxd0`` is provided.
-
-        fluxd0 : float, optional
-            Spectral flux density for 0th mag.
-
-        **kwargs :
-            Any other `Efrho.fluxd` keyword argument.
-
-
-        Returns
-        -------
-        mag : float
-
-
-        Examples
-        --------
-        Reverse of Efrho.from_mag test
-        >>> import astropy.units as u
-        >>> from synphot import SpectralElement
-        >>> from sbpy.activity import Efrho
-        >>> bp = SpectralElement.from_file('https://irsa.ipac.caltech.edu/'
-        ... 'data/SPITZER/docs/files/spitzer/redPUtrans.txt', wave_unit='um',
-        ... comment=';')             # doctest: +REMOTE_DATA +IGNORE_OUTPUT
-        >>> efrho = Efrho(3423.67, u.cm)
-        >>> aper = 10000 * u.km
-        >>> eph = {'rh': 1.45 * u.au,
-        ...        'delta': 0.49 * u.au}
-        >>> efrho.mag('vegamag', aper, eph, bandpass=bp)
-        ...                          # doctest: +REMOTE_DATA +IGNORE_OUTPUT
-        10.0                         # doctest: +REMOTE_DATA +FLOAT_CMP
-
-
-        Notes
-        -----
-        See :ref:`sbpy_spectral_standards` for calibration notes.
-
-        Filter names can be found in the `synphot` `documentation
-        <http://synphot.readthedocs.io/en/stable/synphot/bandpass.html#synphot-predefined-filter>`__.
-
-        A discussion of magnitude zero points can be found in the
-        `synphot` `documentation
-        <http://synphot.readthedocs.io/en/latest/synphot/units.html#counts-and-magnitudes>`__.
-
-        """
-
-        if fluxd0 is None and bandpass is None:
-            raise TypeError('One of `bandpass` or `fluxd0` must be provided.')
+            BB = BlackbodySource(T)
+            try:
+                B = BB.observe(wfb, unit=unit)
+            except SinglePointSpectrumError:
+                B = BB(wfb, unit=unit)
+        else:
+            if not (B.unit.is_equivalent(u.W / u.m**2 / u.um)
+                    or B.unit.is_equivalent(u.W / u.m**2 / u.Hz)
+                    or isinstance(B, u.Magnitude)):
+                raise ValueError(
+                    'B must be a magnitude or have units of spectral '
+                    'flux density, e.g., W/m2/μm or W/m2/Hz')
 
-        efrho0 = Efrho.from_mag(0, unit, aper, eph, bandpass=bandpass,
-                                fluxd0=fluxd0, Tscale=Tscale, **kwargs)
-        return -2.5 * np.log10(self.cm / efrho0.cm)
+        return B
 
 
 class Syndynes:
```

Comment:
The `__init__` method should raise `NotImplementedError` instead of `NotImplemented`